File: backend_entrenemos/src/main/java/com/entrenemos/services/EjercicioService.java
package com.entrenemos.services;

import com.entrenemos.dto.EjercicioDto;

import java.util.List;

public interface EjercicioService {
    EjercicioDto crear(EjercicioDto ejercicioDto);
    List<EjercicioDto> listar();
    EjercicioDto obtenerPorId(Long id);
    EjercicioDto actualizar(Long id, EjercicioDto ejercicioDto);
    void eliminar(Long id);
}

File: frontend_entrenemos/src/app/core/services/ejercicio.service.ts
import { Injectable, inject } from "@angular/core";
import { HttpClient } from "@angular/common/http";
import { Observable } from "rxjs";
import { API_BASE_URL } from "../../config/api.config";
import { Ejercicio } from "../../shared/models/ejercicio.model";

@Injectable({
  providedIn: "root",
})
export class EjercicioService {
  private http = inject(HttpClient);
  private apiUrl = `${API_BASE_URL}/ejercicios`;

  crear(ejercicio: Ejercicio): Observable<Ejercicio> {
    return this.http.post<Ejercicio>(this.apiUrl, ejercicio);
  }

  listar(): Observable<Ejercicio[]> {
    return this.http.get<Ejercicio[]>(this.apiUrl);
  }

  obtenerPorId(id: number): Observable<Ejercicio> {
    return this.http.get<Ejercicio>(`${this.apiUrl}/${id}`);
  }

  actualizar(id: number, ejercicio: Ejercicio): Observable<Ejercicio> {
    return this.http.put<Ejercicio>(`${this.apiUrl}/${id}`, ejercicio);
  }

  eliminar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
